src/faucet.e2e.ts

```diff
@@ -55,39 +69,85 @@ describe("Faucet E2E", () => {
+    const userAddress = randomAddress();
+    const initialBalance = await getUserBalance(userAddress);
 
     await postMessage(`!drip ${userAddress}`);
 
     await until(async () => (await getLatestMessage()).sender === "@bot:parity.io", 500, 10, "Bot did not reply.");
     const botMessage = await getLatestMessage();
     expect(botMessage.body).toContain("Sent @user:parity.io 10 UNITs.");
-    await until(async () => (await getUserBalance(userAddress)).gtn(0), 500, 15, "balance did not increase.");
+    await until(
+      async () => (await getUserBalance(userAddress)).gt(initialBalance),
+      1000,
+      15,
+      "balance did not increase.",
+    );
   });
 
-  test("The bots teleports to a given address", async () => {
-    const userAddress = "1useDmpdQRgaCmkmLFihuw1Q4tXTfNKaeJ6iPaMLcyqdkoS"; // Random address.
+  test("The bot teleports to a given address", async () => {
+    const userAddress = randomAddress();
+    const initialBalance = await getUserBalance(userAddress, parachainApi);
 
-    await postMessage(`!drip ${userAddress}:1001`);
+    await postMessage(`!drip ${userAddress}:${PARACHAIN_ID}`);
 
     await until(async () => (await getLatestMessage()).sender === "@bot:parity.io", 500, 10, "Bot did not reply.");
     const botMessage = await getLatestMessage();
     expect(botMessage.body).toContain("Sent @user:parity.io 10 UNITs.");
+
+    await until(
+      async () => (await getUserBalance(userAddress, parachainApi)).gt(initialBalance),
+      1000,
+      30,
+      "balance did not increase.",
+    );
+  });
+
+  test("The web endpoint responds to a balance query", async () => {
+    const result = await webEndpoint.get("/balance");
+
+    expect(result.status).toEqual(200);
+    expect("balance" in result.data).toBeTruthy();
+    expect(new BN(result.data.balance).gtn(0)).toBeTruthy();
   });
 
   test("The web endpoint drips to a given address", async () => {
-    const userAddress = "5CihnmBEPDRUgZBHAP26YKpcQNd2ATdEWQymsF5jJFci3pnt"; // Random address.
-    expect((await getUserBalance(userAddress)).eqn(0)).toBeTruthy();
+    const userAddress = randomAddress();
+    const initialBalance = await getUserBalance(userAddress);
 
     const result = await webEndpoint.post("/drip/web", { address: userAddress, recaptcha: "anything goes" });
 
     expect(result.status).toEqual(200);
     expect("hash" in result.data).toBeTruthy();
-    await until(async () => (await getUserBalance(userAddress)).gtn(0), 500, 15, "balance did not increase.");
+    await until(
+      async () => (await getUserBalance(userAddress)).gt(initialBalance),
+      500,
+      15,
+      "balance did not increase.",
+    );
+  });
+
+  test("The web endpoint teleports to a given address", async () => {
+    const userAddress = randomAddress();
+    const initialBalance = await getUserBalance(userAddress, parachainApi);
+
+    const result = await webEndpoint.post("/drip/web", {
+      address: userAddress,
+      recaptcha: "anything goes",
```

Comment:
This is using the public testing captcha, right?

Reply:
Correct, it's configured [here](https://github.com/paritytech/substrate-matrix-faucet/blob/3178c07e1788f86c3c46dcd58aafacfdc0f1502b/e2e/bootstrap.sh#L70).